spec/inputs/s3_spec.rb

```diff
@@ -69,6 +69,21 @@
       expect(config.list_new_files).to eq([present_object.key])
     end
 
+    it 'should ignore file if the file match the prefix' do
+        prefix = 'mysource/'
+
+        objects_list = [
+          double(:key => prefix, :last_modified => Time.now),
+          present_object
+        ]
+
+        allow_any_instance_of(AWS::S3::ObjectCollection).to receive(:with_prefix).with(prefix) { objects_list }
+
+        config = LogStash::Inputs::S3.new(settings.merge({ 'prefix' => prefix }))
```

Comment:
This apply to all the ```Logstash::Input::S3``` reference here, we could then redefine the settings merge somehow.